### PR TITLE
ak/clinic_list/normalize.py

### LINKED ISSUE #517: normalize ak/clinic_list
[learn our pipeline: normalize](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#normalize)

Blocked by #497

Normalize existing [`.ndjson`](http://ndjson.org/) records into [the Vaccinate The States data schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Normalized-Location-Schema).

Read all files in the directory passed as the second argument (`sys.argv[2]`), transform each record into the normalized schema (filling in as much detail as possible), then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the normalize stage and how to get set up for development!

### Tips

1. Fetch and parse data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch ak/clinic_list
    poetry run vaccine-feed-ingest parse ak/clinic_list
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest normalize ak/clinic_list
    ```

### Example
[Parsed files for `ak/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/parsed) are converted to [normalized files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/normalized)

### PR DESCRIPTION
**Pull request by rossry**
# 

| Key Details |
|-|
| Resolves #517 |
State: ak |
Site: clinic_list |

## Notes
* none

## Data sample

```json
{"id":"ak_clinic_list:85b07ac7-c9b6-4d09-83ec-a720998c2fbf","name":"North Slope Borough Public Health Nursing-Wellness","address":{"street1":"579 Kongosak St","street2":null,"city":"Barrow","state":"AK","zip":"99723"},"location":{"latitude":71.29069998454801,"longitude":-156.7920099649541},"contact":[{"contact_type":null,"phone":"(907) 852-0270","website":null,"email":null,"other":null}],"languages":null,"opening_dates":null,"opening_hours":null,"availability":{"drop_in":false,"appointments":null},"inventory":[{"vaccine":"pfizer_biontech","supply_level":null}],"access":null,"parent_organization":null,"links":null,"notes":["NSB PHN Wellness clinic, please call for appointments 907-852-0270"],"active":null,"source":{"source":"ak_clinic_list","id":"85b07ac7-c9b6-4d09-83ec-a720998c2fbf","fetched_from_uri":"https://anchoragecovidvaccine.org/providers/","fetched_at":"2021-05-13T15:30:38.805213","published_at":"2021-03-31T22:11:20.475000+00:00","data":{"attributes":{"CreationDate":1608936294888,"Creator":"","EditDate":1617228680475,"Editor":"kelly.isham_Alaska_DHSS","address":"579 Kongosak St","adminContactName":"Andrey Boskhomdzhiev\nNSB PHN Program Coordinator \n907-852-0270\nandrey.boskhomdzhiev@north-slope.org\n","adminEmail":"andrey.boskhomdzhiev@north-slope.org","city":"Barrow","cost_other":null,"datesubmited":1608930000000,"flu_acceptsInsurance":"yes_please_contact_your_insuran","flu_cost":"no_cost","flu_vaccinations":"pfizer","flu_walkins":"no_please_make_an_appointment","globalid":"85b07ac7-c9b6-4d09-83ec-a720998c2fbf","layer_id":0,"objectid":9,"phone":"907-852-0270","publicEmail":null,"publicNotes":"NSB PHN Wellness clinic, please call for appointments 907-852-0270","publicWebsite":null,"service_item_id":"7b65a9e852524227be91ad45d613da33","vaccinationSite":"North Slope Borough Public Health Nursing-Wellness","zipcode":"99723"},"geometry":{"spatialReference":{"latestWkid":4326,"wkid":4326},"x":-156.7920099649541,"y":71.29069998454801}}}}
{"id":"ak_clinic_list:0ed786e2-f009-4919-803a-8addbea8f9bd","name":"Norton Sound Health Corporation","address":{"street1":"228 Greg Kruschek Ave","street2":null,"city":"Nome","state":"AK","zip":"99762"},"location":{"latitude":64.49934297442883,"longitude":-165.3783630383781},"contact":[{"contact_type":null,"phone":"(907) 434-0461","website":null,"email":null,"other":null},{"contact_type":null,"phone":null,"website":null,"email":"pharmacyrx@nshcorp.org","other":null},{"contact_type":null,"phone":null,"website":"https://www.picktime.com/NSHC","email":null,"other":null}],"languages":null,"opening_dates":null,"opening_hours":null,"availability":{"drop_in":false,"appointments":null},"inventory":[{"vaccine":"pfizer_biontech","supply_level":null},{"vaccine":"moderna","supply_level":null}],"access":null,"parent_organization":null,"links":null,"notes":["Open 9 AM -6 PM Monday through Friday, 10 AM - 2PM Saturday."],"active":null,"source":{"source":"ak_clinic_list","id":"0ed786e2-f009-4919-803a-8addbea8f9bd","fetched_from_uri":"https://anchoragecovidvaccine.org/providers/","fetched_at":"2021-05-13T15:30:38.805213","published_at":"2021-02-17T23:45:05.617000+00:00","data":{"attributes":{"CreationDate":1609022652065,"Creator":"","EditDate":1613605505617,"Editor":"kelly.isham_Alaska_DHSS","address":"228 Greg Kruschek Ave","adminContactName":"Alex Ogren aogren@nshcorp.org\nHeather Lefebvre hlefebvre@nshcorp.org\nMark Peterson mgpeterson@nshcorp.org","adminEmail":"aogren@nshcorp.org","city":"Nome","cost_other":null,"datesubmited":1609016400000,"flu_acceptsInsurance":"no","flu_cost":"no_cost","flu_vaccinations":"moderna,pfizer","flu_walkins":"no_please_make_an_appointment","globalid":"0ed786e2-f009-4919-803a-8addbea8f9bd","layer_id":0,"objectid":10,"phone":"907-434-0461","publicEmail":"pharmacyrx@nshcorp.org","publicNotes":"Open 9 AM -6 PM Monday through Friday, 10 AM - 2PM Saturday.","publicWebsite":"https://www.picktime.com/NSHC","service_item_id":"7b65a9e852524227be91ad45d613da33","vaccinationSite":"Norton Sound Health Corporation","zipcode":"99762"},"geometry":{"spatialReference":{"latestWkid":4326,"wkid":4326},"x":-165.3783630383781,"y":64.49934297442883}}}}
{"id":"ak_clinic_list:7840e505-ff68-4400-8381-eb67954c43c6","name":"Soldotna Professional Pharmacy","address":{"street1":"299 N Binkley St","street2":null,"city":"Soldotna","state":"AK","zip":"99669"},"location":{"latitude":60.49456000351094,"longitude":-151.07297003784822},"contact":[{"contact_type":null,"phone":"(907) 262-3800","website":null,"email":null,"other":null},{"contact_type":null,"phone":null,"website":"https://cw2-alaska-production.herokuapp.com/clinic/search","email":null,"other":null}],"languages":null,"opening_dates":null,"opening_hours":null,"availability":{"drop_in":false,"appointments":null},"inventory":[{"vaccine":"pfizer_biontech","supply_level":null},{"vaccine":"moderna","supply_level":null}],"access":null,"parent_organization":null,"links":null,"notes":null,"active":null,"source":{"source":"ak_clinic_list","id":"7840e505-ff68-4400-8381-eb67954c43c6","fetched_from_uri":"https://anchoragecovidvaccine.org/providers/","fetched_at":"2021-05-13T15:30:38.805213","published_at":"2021-03-31T22:14:22.480000+00:00","data":{"attributes":{"CreationDate":1609099691362,"Creator":"","EditDate":1617228862480,"Editor":"kelly.isham_Alaska_DHSS","address":"299 N Binkley St","adminContactName":"Justin Ruffridge","adminEmail":"justin@inletrxgroup.com","city":"Soldotna","cost_other":null,"datesubmited":1609102800000,"flu_acceptsInsurance":"yes_please_contact_your_insuran","flu_cost":"no_cost","flu_vaccinations":"moderna,pfizer","flu_walkins":"no_please_make_an_appointment","globalid":"7840e505-ff68-4400-8381-eb67954c43c6","layer_id":0,"objectid":11,"phone":"907-262-3800","publicEmail":null,"publicNotes":null,"publicWebsite":"https://cw2-alaska-production.herokuapp.com/clinic/search","service_item_id":"7b65a9e852524227be91ad45d613da33","vaccinationSite":"Soldotna Professional Pharmacy","zipcode":"99669"},"geometry":{"spatialReference":{"latestWkid":4326,"wkid":4326},"x":-151.07297003784822,"y":60.49456000351094}}}}
{"id":"ak_clinic_list:840db74d-4eae-4802-84f8-37bcee07fd68","name":"Dr. deSchweinitz, Dr. Trujillo, ANP Carson-McCollum","address":{"street1":"9500 Independence Drive Suite 700","street2":null,"city":"Anchorage","state":"AK","zip":"99507"},"location":{"latitude":61.13475074651249,"longitude":-149.8419484061434},"contact":[{"contact_type":null,"phone":"(907) 569-3600","website":null,"email":null,"other":null},{"contact_type":null,"phone":null,"website":null,"email":"frontdesk@akdocs.com","other":null},{"contact_type":null,"phone":null,"website":"http://www.akdocs.com/","email":null,"other":null}],"languages":null,"opening_dates":null,"opening_hours":null,"availability":{"drop_in":false,"appointments":null},"inventory":[{"vaccine":"pfizer_biontech","supply_level":null}],"access":null,"parent_organization":null,"links":null,"notes":["Complete the COVID vaccination registration form located on the website, http://www.akdocs.com. Email the completed form and copies of picture ID and insurance/medicare card(s) to frontdesk@akdocs.com.  Our office will call you to set up an appointment."],"active":null,"source":{"source":"ak_clinic_list","id":"840db74d-4eae-4802-84f8-37bcee07fd68","fetched_from_uri":"https://anchoragecovidvaccine.org/providers/","fetched_at":"2021-05-13T15:30:38.805213","published_at":"2021-02-08T20:34:34.040000+00:00","data":{"attributes":{"CreationDate":1609112999158,"Creator":"","EditDate":1612816474040,"Editor":"kelly.isham_Alaska_DHSS","address":"9500 Independence Drive Suite 700","adminContactName":"Crystal Cornejo, CMA\nElizabeth deSchweinitz, MD\nDwayne Trujillo, MD","adminEmail":"practicemanager@akdocs.com","city":"Anchorage","cost_other":"Administration fee will be billed to insurance.","datesubmited":1609102800000,"flu_acceptsInsurance":"yes_please_contact_your_insuran","flu_cost":"other","flu_vaccinations":"pfizer","flu_walkins":"no_please_make_an_appointment","globalid":"840db74d-4eae-4802-84f8-37bcee07fd68","layer_id":0,"objectid":12,"phone":"907-569-3600","publicEmail":"frontdesk@akdocs.com","publicNotes":"Complete the COVID vaccination registration form located on the website, http://www.akdocs.com. Email the completed form and copies of picture ID and insurance/medicare card(s) to frontdesk@akdocs.com.  Our office will call you to set up an appointment.","publicWebsite":"http://www.akdocs.com","service_item_id":"7b65a9e852524227be91ad45d613da33","vaccinationSite":"Dr. deSchweinitz, Dr. Trujillo, ANP Carson-McCollum","zipcode":"99507"},"geometry":{"spatialReference":{"latestWkid":4326,"wkid":4326},"x":-149.8419484061434,"y":61.13475074651249}}}}
{"id":"ak_clinic_list:2d4e9ef7-401e-42b4-81be-e5f6752cd3ef","name":"The Children's Clinic","address":{"street1":"4001 Dale St","street2":null,"city":"Anchorage","state":"AK","zip":"99508"},"location":{"latitude":61.18399500901996,"longitude":-149.81748499433394},"contact":[{"contact_type":null,"phone":"(907) 562-2944","website":null,"email":null,"other":null},{"contact_type":null,"phone":null,"website":null,"email":"info@tccpeds.com","other":null},{"contact_type":null,"phone":null,"website":"https://myhealth.alaska.gov/clinic/search","email":null,"other":null}],"languages":null,"opening_dates":null,"opening_hours":null,"availability":{"drop_in":false,"appointments":null},"inventory":[{"vaccine":"moderna","supply_level":null}],"access":null,"parent_organization":null,"links":null,"notes":["Drive-thru vaccine clinic"],"active":null,"source":{"source":"ak_clinic_list","id":"2d4e9ef7-401e-42b4-81be-e5f6752cd3ef","fetched_from_uri":"https://anchoragecovidvaccine.org/providers/","fetched_at":"2021-05-13T15:30:38.805213","published_at":"2021-01-11T22:51:49.862000+00:00","data":{"attributes":{"CreationDate":1609171433841,"Creator":"","EditDate":1610405509862,"Editor":"kelly.isham_Alaska_DHSS","address":"4001 Dale St","adminContactName":"Deyana Thayer, Grace Stout","adminEmail":"dthayer@tccpeds.com","city":"Anchorage","cost_other":"Administration fee will be billed to insurance only, no cost to patient","datesubmited":1609189200000,"flu_acceptsInsurance":"yes_please_contact_your_insuran","flu_cost":"other","flu_vaccinations":"moderna","flu_walkins":"no_please_make_an_appointment","globalid":"2d4e9ef7-401e-42b4-81be-e5f6752cd3ef","layer_id":0,"objectid":15,"phone":"907-562-2944","publicEmail":"info@tccpeds.com","publicNotes":"Drive-thru vaccine clinic","publicWebsite":"https://myhealth.alaska.gov/clinic/search","service_item_id":"7b65a9e852524227be91ad45d613da33","vaccinationSite":"The Children's Clinic","zipcode":"99508"},"geometry":{"spatialReference":{"latestWkid":4326,"wkid":4326},"x":-149.81748499433394,"y":61.18399500901996}}}}
{"id":"ak_clinic_list:14ab4a33-3204-42fe-8d46-6e572d094224","name":"Costco NE Anchorage","address":{"street1":"4125 Debarr Rd","street2":null,"city":"Anchorage","state":"AK","zip":"99508"},"location":{"latitude":61.21115799118353,"longitude":-149.80381796593025},"contact":[{"contact_type":null,"phone":null,"website":"https://www.costco.com/covid-vaccine.html","email":null,"other":null}],"languages":null,"opening_dates":null,"opening_hours":null,"availability":{"drop_in":false,"appointments":null},"inventory":[{"vaccine":"moderna","supply_level":null}],"access":null,"parent_organization":{"id":"costco","name":null},"links":null,"notes":["Clinics will be reserved for COVID vaccination only"],"active":null,"source":{"source":"ak_clinic_list","id":"14ab4a33-3204-42fe-8d46-6e572d094224","fetched_from_uri":"https://anchoragecovidvaccine.org/providers/","fetched_at":"2021-05-13T15:30:38.805213","published_at":"2021-04-21T01:22:01.185000+00:00","data":{"attributes":{"CreationDate":1609180437466,"Creator":"","EditDate":1618968121185,"Editor":"kelly.isham_Alaska_DHSS","address":"4125 Debarr Rd","adminContactName":"Becky Dant, PharmD\n425-416-5577","adminEmail":"rdant@costco.com","city":"Anchorage","cost_other":null,"datesubmited":1609185600000,"flu_acceptsInsurance":"yes_please_contact_your_insuran","flu_cost":"no_cost","flu_vaccinations":"moderna","flu_walkins":"no_please_make_an_appointment","globalid":"14ab4a33-3204-42fe-8d46-6e572d094224","layer_id":0,"objectid":16,"phone":null,"publicEmail":null,"publicNotes":"Clinics will be reserved for COVID vaccination only","publicWebsite":"https://www.costco.com/covid-vaccine.html","service_item_id":"7b65a9e852524227be91ad45d613da33","vaccinationSite":"Costco NE Anchorage","zipcode":"99508"},"geometry":{"spatialReference":{"latestWkid":4326,"wkid":4326},"x":-149.80381796593025,"y":61.21115799118353}}}}
```

## Before Opening a PR
- [x] I tested this using the CLI (e.g., `poetry run vaccine-feed-ingest <state>/<site>`)
- [x] I ran auto-formatting: `poetry run tox -e lint-fix`

### CODE REVIEW

File: vaccine_feed_ingest/runners/ak/clinic_list/normalize.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+logger = getLogger(__file__)
+
+
+def _get_id(site: dict) -> str:
+    return site["attributes"]["globalid"]
+
+
+def _get_contacts(site: dict) -> Optional[List[schema.Contact]]:
+    ret = []
+    if phone := site["attributes"]["phone"]:
```
Comment (bryanculbertson) [inline vaccine_feed_ingest/runners/ak/clinic_list/normalize.py:28]: Are you sure the fields always exist? If not then use `.get` `site["attributes"].get("phone")`

**Review by bryanculbertson — APPROVED** [review]

**Comment by bryanculbertson** [comment]
Approved and merged.

Can you take a 2nd pass and make sure that the fields are guaranteed to be present? And if not then add more checks, or use `.get()`

File: vaccine_feed_ingest/runners/ak/clinic_list/normalize.py (hunk 1 above)
Comment (rossry) [reply vaccine_feed_ingest/runners/ak/clinic_list/normalize.py:28]: It exists in 146/146 of the existing records and sometimes is null; I assume that that's sufficient proof that it's being enforced in the upstream schema?

**Comment by rossry** [comment]
As in the subthread -- the fields are sometimes present but nulled (and are present in 146/146 cases), which I'd take as evidence that they're guaranteed to exist in the upstream schema.

**Pull request merged**